### PR TITLE
docs: add google integration

### PR DESCRIPTION
**Pull request by viniciossilva3**
- Added documentation teaching how to configure an application using Google's OAuth API to use the components present in Langflow. Example: (Google Drive Loader, Google Drive Search and Gmail Loader).

### CODE REVIEW

**Comment by mendonk** [comment]
@viniciossilva3 Thanks for this contribution, awesome! I pushed some style updates, but the content looks good to me.

**Review by mendonk — APPROVED** [review]
Pushed some style changes, but the content LGTM

**Comment by ogabrielluiz** [comment]
Holding this off until we release the components.

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+- **Google Drive Search**
+   Searches Google Drive files using provided credentials and query parameters.
+
+To set up Google OAuth, create an OAuth app in Google Cloud, obtain the necessary credentials and access tokens, and add them to Langflow’s Google components.
+
+## Create an OAuth Application in Google Cloud {#5b8981b15d86192d17b0e5725c1f95e7}
+
+1. Navigate to the [Google Cloud Console](https://console.cloud.google.com/).
+
+2. Click **Select a project** to choose an existing projec, or create a new project.
```
Comment (mendonk) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:27]:
```suggestion
2. Click **Select a project** to choose an existing project, or create a new project.
```

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+---
+title: "Setup Google OAuth for Langflow Integration"
```
Comment (mendonk) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:2]:
```suggestion
title: "Integrate Google OAuth with Langflow"
```

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+## Retrieve Access and Refresh Tokens
+
+With your OAuth application configured and your credentials JSON file created, follow these steps to authenticate the Langflow application.
+
+1. Create a new project in Langflow.
+2. Add a **Google OAuth Token** component to your flow.
+3. Input in the field **Credentials File** on the Google OAuth Token component, the JSON file containing the Client ID credentials you downloaded from Google in the [previous steps](#5b8981b15d86192d17b0e5725c1f95e7).
+4. Click the **Play** button in the **Google OAuth Token** component to authenticate your application.
+When the component code is executed, a new tab may open in the browser to authenticate your application using your Google Cloud account. You must authenitcate with the account where you created the OAuth credentials for your application.
```
Comment (mendonk) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:55]:
```suggestion
When the component code is executed, a new tab should open in the browser to authenticate your application using your Google Cloud account. You must authenticate with the account where you created the OAuth credentials for your application.
```

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+If a new tab does not open automatically, check the Langflow **Logs** for the Google authentication URL. Open this URL in your browser to complete the authentication. Only after authenticating will the JSON token be generated.
+
+5. After successful authentication, your Langflow application can now request and refresh tokens for your application. These tokens enable Langflow to interact with Google services on your behalf and execute the requests you’ve specified.
+By default, token validity is managed by Google’s servers. In Langflow, tokens refresh automatically after initial authentication. However, if your application is inactive for an extended period, the tokens may expire, requiring you to re-authenticate to resume use in Langflow.
+
+## Configure Google Components in Langflow
+
+In this example, use the **Google Drive Loader** component to load a text file hosted on Google Drive, translate the text to Spanish, and return it to a chat output.
+
+For a pre-built example of this flow, download this JSON file to your local machine.
```
Comment (mendonk) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:65]:
```suggestion
For a pre-built example of this flow, download the <a href="./files/Google_Drive_Docs_Translations_Example.json" download>Google Drive Document Translation Example Flow JSON</a> to your local machine.
```

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+- Flow Google Drive Docs Translations Example -
```
Comment (mendonk) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:67]:
```suggestion
```

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  (<a href="./files/Google_Drive_Docs_Translations_Example.json" download>Download link</a>)
```
Comment (mendonk) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:68]:
```suggestion
```

**Review by mendonk — CHANGES_REQUESTED** [review]
More style items for review

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md (hunk 6 above)
Comment (viniciossilva3) [reply docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md]: Hi @mendonk Could you check the changes? https://github.com/langflow-ai/langflow/pull/4382/commits/0ae39d163900b731433691ab45cdbdeb3676cd24

**Comment by viniciossilva3** [comment]
Hi @mendonk Could you check the changes? https://github.com/langflow-ai/langflow/pull/4382/commits/0ae39d163900b731433691ab45cdbdeb3676cd24

**Comment by mendonk** [comment]
@viniciossilva3 
Reviewed for style and testing - 
1. I can't get my credentials JSON file to load so I can't confirm this flow works.
2. What kind of file should the user load from their drive?
3. The Google process for OAuth client IDs is changing soon, so I updated the gifs and the steps.
4. Can you get a working gif of this flow?
Screenshot 2024-11-14 at 11 23 04 AM

**Comment by viniciossilva3** [comment]
> @viniciossilva3 Reviewed for style and testing -
> 
> 1. I can't get my credentials JSON file to load so I can't confirm this flow works.
> 2. What kind of file should the user load from their drive?
> 3. The Google process for OAuth client IDs is changing soon, so I updated the gifs and the steps.
> 4. Can you get a working gif of this flow?
>    Screenshot 2024-11-14 at 11 23 04 AM

Hi @mendonk! To use OAuth, you need to create a consent screen for the application containing the permissions that will be allowed, and also an OAuth Client ID.
And when you use it, you need to send the URLs with the scopes requesting the permissions to the application.

It would be nice if there was a Langflow consent screen so we could connect directly using the UI. Users would only need to log in to Langflow and use it, but currently this function does not exist yet.

Can you recommend a tool to convert it to gif? The ones I tested here had very poor quality...

An example of this flow working.


https://github.com/user-attachments/assets/e3849170-d5e0-4438-8682-910f16ca2692

**Comment by mendonk** [comment]
@viniciossilva3 thanks for the followup!
I still can't get my json file to load in langflow, but I see you have tested it and it works.
I've made my changes and this will now go on to next reviewer.

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 7]
```diff
@@ -1,165 +1,69 @@
...
-   - Ensure you **publish** the app if it’s not restricted to internal use.
-
-![OAuth Consent Screen](/img/google/setup-oauth-consent-screen.png)
-
-:::info
-
-- Configuring the OAuth consent screen is crucial for obtaining user permissions.
-
-::::
+3. To enable APIs for the project, select **APIs & Services** and click **Library**. Enable the APIs you need for your project. For example, if your flow uses the Google Drive component, enable the Google Drive API.
```
Comment (KimberlyFields) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:22]:
```suggestion
3. To enable APIs for the project, select **APIs & Services**, and then click **Library**. Enable the APIs you need for your project. For example, if your flow uses the Google Drive component, enable the Google Drive API.
```

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 8]
```diff
@@ -1,165 +1,69 @@
...
-1. **Authenticate the Application**
-
-   - Create a new project in Langflow.
-   - Add a Google OAuth Token component.
-   - Input in the field **Credentials File** on the Google OAuth Token component, the JSON file containing the Client ID credentials you downloaded from Google in the [previous steps](#5b8981b15d86192d17b0e5725c1f95e7).
-   - Run the Google OAuth Token component to authenticate your application.
+1. Create a new project in Langflow.
+2. Add a **Google OAuth Token** component to your flow.
+3. In the **Credentials File** field of the Google OAuth Token component, enter the path to your **Credentials File**, the JSON file containing the Client ID credentials you downloaded from Google in the previous steps.
+4. To authenticate your application, in the **Google OAuth Token** component, click the **Play** button.
```
Comment (KimberlyFields) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:40]:
```suggestion
4. To authenticate your application, in the **Google OAuth Token** component, click **Play**.
```

File: docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md [hunk 9]
```diff
@@ -1,165 +1,69 @@
...
----
-
-By following these steps, your Langflow environment will be fully integrated with Google services, providing a powerful tool for automating workflows that involve Google Gmail, Drive, and more.
+5. To allow the Langflow component to access the file in Google Drive, copy the Google Drive File ID from the document's URL.
+:::info
+The file ID is located between `/d/` and `/edit` in a Google Drive document's URL.
+For example, in the URL `https://drive.google.com/file/d/1a2b3c4D5E6F7gHI8J9klmnopQ/edit`, the File ID is `1a2b3c4D5E6F7gHI8J9klmnopQ`.
+:::
+6. In the Google Drive loader component, in the **Document ID** field, paste the document URL.
+7. Click the **Chat output** component, click the **Play** button.
```
Comment (KimberlyFields) [inline docs/docs/Integrations/Google/integrations-setup-google-oauth-langflow.md:68]:
```suggestion
7. Click the **Chat output** component, and then click **Play**.
```

**Review by KimberlyFields — APPROVED** [review]

**Review by mendonk — APPROVED** [review]
Pushed some changes, lgtm

**Pull request merged**